lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -173,6 +175,11 @@ def initialize(config_root, config_path, loaded_config)
     @data_providers = nil
   end
 
+  def fail(issue, args = EMPTY_HASH, line = nil)
```

Comment:
We use this pattern elsewhere in the 4.x codebase (pops) - when that was added I was unaware that `fail` was an alias for `raise`. 
It is probably worth to modify that everywhere (new ticket) and use some other name for this - `raise_issue` or somesuch perhaps?